### DISCUSSION TITLE
백엔드 개발자와의 협업, 어떻게 하고 있나요?

### DISCUSSION THREAD

**Discussion by woowapark** [woowacourse/react-subway-map#2] [General]
- 프론트 크루들끼리만 논의할 때와 다른 부분이 있었나요?
- 논의하는 과정에서 이슈는 없었나요? 서로 어떤 부분을 논의하였나요?
- 그 외 나누고 싶은 이야기, 경험

**Comment by yujo11** [comment]
### 파노 그루밍

**어려웠던 부분**

- 예외사항을 분류하면서 어떻게 소통할 수 있을까를 고민했는데 소통하는 방법보다는 문제를 식별하는게 우선.
- 식별할 수 없는 경우에 대해서는 에러 메세지를 응답해주는게 좋지 않겠냐
- RFC문서에 에러에 대한 코드를 통일해서 백엔드와 프론트가 약속으로 갖고 있음 좋다.

**달랐던 부분**

- 예외처리를 뭉뚱그려서 했던거 같은데 그런거에 대한 고지가 명세에 고정되어 있었다.

---

### 피터 서니

**어려웠던 부분**

- 서로의 도메인에 대한 배경지식이 없다보니 막상 프로젝트를 하고 기능을 구현하는 과정에서 다시 부랴부랴 연락하게 되는 경우가 많다.

**달랐던 부분**

- 에러 처리
- 평소에 신경을 안 썼던 데이터를 어떤 식으로 주가 될건지, 엔드 포인트를 정하는 부분에 대해서 얘기,
- 구간 데이터 같은 경우는 배열로만 보내줬는데 → upstation id, downstation id 오도록 변경
- 백엔드 크루 중 한명이 먼저 Restfull 한 데이터를 보내주기로 했다.
- 몰랐던 부분에 대해서 많이 알게 됐다.

---

### 유조

**어려웠던 부분**

- 서로의 도메인에 대한 이해가 떨어짐

**달랐던 점**

- 기존에는 주어진 API를 변경할 수가 없는 상태에서 미션을 진행했기 때문에 사고가 클라이언트 단에서만 한정되어 있었다. 하지만 이번 미션에서는 논의를 통해 짜여진 API의 response, response body, status code등을 변경할 수 있어 더 넓은 범위로 고민하고 선택할 수 있게 됐다.

**Comment by Puterism** [comment]
### 주모

- 이슈가 많이 없어서 프론트 크루들과 협업하는 과정과 비슷했다.
- 서로 다른 기초 지식이 다른 부분이 있는 부분에 대해서 서로 이야기를 나누었음
- 요구사항에 대해 얼마나 부담이 될 지 파악이 잘 안 돼서 요구하기 어려울 때가 있음

### 브랜

- 프론트 입장에서는 있었으면 하는 기능들을 요청함
- 필요한 상황에서 만들어 달라고 요청하면 시간이 걸리기 때문에, 무엇이 필요할 지 미리 파악하고 말하면 좋은데, 이를 파악하는 것이 어려웠음.
- 어떤 예외가 발생했는지 프론트에서 파악하고 분류할 수 있도록 이야기함

### 체프

- 얼핏 봤을 때는 큰 문제가 없어보였는데, 생각과 다르게 동작하는 부분이 많았다.
- API를 처음부터 끝까지 한번 살펴보는 과정이 많이 도움이 되었습니다. 이 과정에서 데이터를 어떻게 받아올 지, 에러는 어떻게 반환할 지, 에러 메시지는 어떻게 받아올 지 의논했다.
- 프론트에서 코드로 API를 호출할 때, 어떤 식으로 가져오게 될 지 생각하는 게 쉽지 않았다.

### 하루

- 백엔드 분들이 사용하시는 용어를 못 알아들을 때가 있었다. (TLS?)
- 협업용 노션 페이지를 작성해서 내용을 공유하고 회의를 진행했다. 회의를 진행하기 전에 페어인 티케와 미리 백엔드 분들에게 요구하고 싶은 항목을 정리했다.
    - (전체보기 필요한 json, lines GET 요청 시 노선에 필요한 것만 보내주도록 변경, email 중복 API 요청, 에러 발생 시 body에 message담아주기, 상태코드 정리, accessToken 없이 요청시 에러 처리)
- 백엔드 분들께 요구사항과 그게 필요한 이유를 UI를 보여드리면서 말씀드렸더니 흔쾌히 들어주셔서 필요한 사항을 모두 받을 수 있게 되었다.

**Comment by 2SOOY** [comment]
### 곤이
- 직접 만나서 회의를 진행 ⇒ 백엔드 분들이 편의를 많이 봐줌
- 백엔드에게 실제 원하는 것을 요청
    - 노선 정보에 구간 정보가 들어있고, 구간 정보는 별도로 제공하지 않음
    - 전체 보기 기능 ⇒ `환승역`에 대한 정보도 포함했으면 좋겠다.
    - 이중 배열 사용없이 1번의 반복문 만으로 원하는 정보를 얻어올 수 있도록
- 백엔드와 프론트간의 명세를 확인하는 작업 ⇒ 문제점을 잘 파악

### 지그
- 이제 기능 작업에 들어가기 시작해서 회의를 진행할 예정
    - 백엔드 측에서 적절한 에러 메시지를 전달해줬으면 좋겠다
    - 환승역에 대한 정보는 어떻게 요청할지 고민.
    - 이메일 중복 체크

### 동동
- 협업 방식 → 노션페이지 + 깃허브 위키 + slack을 통한 실시간 대화
- 네트워크관련 지식이 부족하다보니 무언가를 요청하기가 꺼렸음
    - 조금씩 편하게 이야기 할 수 있도록 해야겠다.
- 요청할 때의 태도는 부드럽게 해야겠다는 생각이 들었음

### 도비
- 구글 meet을 통한 1번의 회의를 진행함
- 백엔드에서 받은 명세에 최대한 맞춰서 개발하기로 결정
- 전체 노선 조회
    - 기존에 있는 lines만으로도 충분히 FE단에서 구현할 수 있음 ⇒ 그냥 명세 그대로 활용하기로함
- 요금 추가
    - 이후 추가된다면 그것에 대한 요청을 추가할 예정

### 콜린
- 이전 레벨 1에서 브라운이 제공해준 명세에 대해 검증하는 작업이 필요했는데, 거의 동일하다고 해서 우선적으로 추가 요청은 없었음
- 우선 백엔드가 준 명세에 맞춰서 개발하려고 했었음
- 크루들의 이야기를 듣고 적절한 추가적인 요청이 필요하다고 생각함
(환승역, 이메일 중복 체크)

**Comment by swon3210** [comment]
### **[ 카일 ]**

어려움보다는 느낀게, 사용하는 단어 등이 서로 달라서 소통이 잘 안되는 경우가 많았다. 소통을 하려면 서로의 언어에 대해서 공부할 필요가 있다고 느낌. 케이스를 분리하고 케이스 마다 이슈를 올릴 수 있는 공간을 만듬. 의사 코드를 통해 소통함. 진척상황을 서로 확인할 수 있는 플랫폼이 있는게 중요함.
### **[ 인치 ]**

각자의 영역이 따로 있다. 서로 그 영역에 대해 모르기 때문에, 모르는 부분에 대해 물어보는 시간이 필요했음. 프로젝트 자체에 대한 이해도 조금씩 다른게 문제였음. 명세 맞추기 등에서 노력이 필요했음.

### **[ 티케 ]**


자바 백엔드를 본인이 이미 해봤기 때문에 뭐를 귀찮아하고, 어디까지 할수 있는지에 대한 감이 있음. 이미 백엔드 분들이 해놓은게 많아서 소통에 어려움이 많지는 않았음. 이를 기반으로 많은 세세한 부분에서의 요청을 드릴 수가 있었음. 서로 적극적으로 상대방에게 제안하는 식의 소통이 많은 도움이 되었음. API 문서를 보면 에러 메세지가 여러 분기로 분리가 되는데, 이 부분에 대해서도 재밌게 이야기 나눠봄.

### **[ 썬 ]**

노션으로 소통을 할 것으로 보임. 아직 회의를 안 해봐서 어떻게 될지는 모르겠지만, 소통을 위한 폼을 미리 만들어두었음. 일정이 서로 빡빡해서 추가 기능을 요청하기는 힘들어짐.


### **[ 크리스 ]**


주로 전체보기 API 를 통해 가져올 데이터 형식에 대해 이야기를 많이 나눔

백엔드에서 잘못된 요청에 대한 처리를 어떻게 할지에 대해서 이야기를 나눔

소통 플랫폼이 있는게 중요함
